app/models/concerns/smooch_messages.rb

```diff
@@ -278,15 +278,60 @@ def bundle_list_of_messages(list, last, reject_payload = false)
       self.adjust_media_type(bundle)
     end
 
-    def handle_bundle_messages(type, list, last, app_id, annotated, send_message = true)
-      bundle = self.bundle_list_of_messages(list, last)
-      if ['default_requests', 'irrelevant_search_result_requests'].include?(type)
-        self.process_message(bundle, app_id, send_message, type)
+    def bundle_list_of_messages_to_items(list, last)
+      # Collect messages from list based on media files, long text and short text
+      # so we have three types of messages
+      # Long text (text with number of words > min_number_of_words_for_tipline_submit_shortcut)
+      # Short text (text with number of words <= min_number_of_words_for_tipline_submit_shortcut)
+      # Media (image, audio, video, etc)
+      messages = []
+      # Define a text variable to hold short text
+      text = []
+      list.collect{ |m| JSON.parse(m) }.sort_by{ |m| m['received'].to_f }.each do |message|
+        if message['type'] == 'text'
+          # Get an item for long text (message that match number of words condition)
+          if message['payload'].nil? && ::Bot::Alegre.get_number_of_words(message['text'].to_s) > CheckConfig.get('min_number_of_words_for_tipline_submit_shortcut', 10, :integer)
+            messages << message
+          else
+            text << begin JSON.parse(message['payload'])['keyword'] rescue message['text'] end
+          end
+        else
+          # Get an item for each media file
+          message['text'] = [message['text'], message['mediaUrl'].to_s].compact.join("\n#{Bot::Smooch::MESSAGE_BOUNDARY}")
+          messages << self.adjust_media_type(message)
+        end
+      end
+      # collect text that did not match min_number_of_words_for_tipline_submit_shortcut(short text)
+      short_text = text.reject{ |t| t.blank? }
+      if messages.blank?
+        # No messages exist (this happens when all messages are short text)
+        # So will create a new message of type text and assign short text to it
+        message = last.clone
+        message['text'] = short_text.join("\n#{Bot::Smooch::MESSAGE_BOUNDARY}")
+        messages << message
+      else
+        # Attach all existing text (media text, long text and short text) to each item
+        # and add a boundary so we can easily split messages if needed
+        all_text = messages.collect{ |m| m['text'] }.concat(short_text).reject{ |t| t.blank? }.join("\n#{Bot::Smooch::MESSAGE_BOUNDARY}")
```

Comment:
Does this put the messages in the correct order? For example, if we have:
Long text
Short text
Media 1

I think this code sets `all_text` to 
Long text
Media 1
Short text

But we want `all_text` to be
Long text
Short text
Media 1

Reply:
@melsawy please review.

Reply:
yes @computermacgyver my code set short text at the end, I'll set it in right order

Reply:
Done